test/sagemaker_tests/pytorch/training/resources/resnet18/main.py

```diff
@@ -120,7 +120,7 @@ def data_generator(self):
                 yield image_np, label
 
         except StopIteration:
-            raise StopIteration
+            return
```

Comment:
This changes the behavior of a test when there is a `StopIteration` Exception. Is this to be expected? Could you explain why this change was made?

Reply:
https://www.python.org/dev/peps/pep-0479/ There was a change introduced to generator behavior in py 3.7+. When StopIteration is raised inside a generator, it is replaced with RuntimeError. Before it used to end the iteration. So, to overcome that, need to return once it reaches end of the iteration